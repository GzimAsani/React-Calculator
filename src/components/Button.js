import PropTypes from 'prop-types';

const Button = ({ name, clickHandler }) => {
  const handleClick = () => clickHandler(name);
  const fetchResult = (name) => {
    switch (name) {
      case '0':
        return 'Button zero';
      case '+':
      case '-':
      case 'x':
      case '/':
      case '=':
        return 'Button operator';
      default:
        return 'Button';
    }
  };
  return (
    <button type="button" onClick={handleClick} className={fetchResult(name)}>
      {name}
    </button>

  );
};

Button.propTypes = {
  name: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired,
};

export default Button;
