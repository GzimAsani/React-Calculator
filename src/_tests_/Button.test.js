import { create } from 'react-test-renderer';
import Button from '../components/Button';

test('render correctly', () => {
  const handleClick = () => { ''; };
  const tree = create(<Button clickHandler={handleClick} name="" />).toJSON();
  expect(tree).toMatchSnapshot();
});
