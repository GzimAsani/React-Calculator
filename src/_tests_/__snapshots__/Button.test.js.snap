// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<button
  onClick={[Function]}
  type="button"
>
  
</button>
`;
